src/index.tsx

```diff
@@ -443,12 +443,19 @@ export class Rnd extends React.Component<Props, State> {
     let x;
     let y;
     const offset = this.getOffsetFromParent();
+    const isOnResizeCallbackDefined = this.props.onResize !== Rnd.defaultProps.onResize;
+    const width = this.resizable.state.original.width + delta.width;
+    const height = this.resizable.state.original.height + delta.height;
     if (/left/i.test(direction)) {
       x = this.state.original.x - delta.width;
       // INFO: If uncontrolled component, apply x position by resize to draggable.
       if (!this.props.position) {
         this.draggable.setState({ x });
       }
+      if (!isOnResizeCallbackDefined) {
+        this.draggable.setState({ x });
+        this.resizable.updateSize({ width, height });
```

Comment:
@regulyarniy Thanks for your great work.

> When using controlled component without onResize 

Even when using `size` and `position`, it will be `uncontolled` under certain conditions.I think this will confuse many users. I'll consider what to do.